ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java

```diff
@@ -471,4 +448,41 @@ public void setFqdn(String fqdn) {
     public void setRestAdapter(RestAdapter restAdapter) {
         this.restAdapter = restAdapter;
     }
+
+    @Override
+    public void postStart() {
+        super.postStart();
+        generatePassword();
+    }
+
+    /**
+     * Need to wait server to start before resetting the password.
+     */
+    protected void generatePassword() {
+        if (Strings.isBlank(getAttribute(AmbariServer.PASSWORD))) {
+            String password;
+            if (Strings.isBlank(getConfig(AmbariServer.PASSWORD))) {
+                password = Identifiers.makeRandomId(10);
```

Comment:
I think we've agreed that this isn't a good way to generate a password.  If password isn't set then I think this should be an error.  That way we still have a default of ambari but if someone deliberately sets it to blank it could fail with a sensible error message.  If someone wants a generated password then brooklyn has a CreatePasswordSensor.